src/modules/cmdpal/extensionsdk/Microsoft.CmdPal.Extensions.Helpers/ListHelpers.cs

```diff
@@ -73,21 +94,27 @@ public static void InPlaceUpdateList<T>(Collection<T> original, Collection<T> ne
             else
             {
                 // it isn't. Add it.
-                original.Insert(i, newContents[i]);
+                original.Insert(i, newItem);
             }
+
+            i++;
         }
 
         // Remove any extra trailing items from the destination
-        while (original.Count > newContents.Count)
+        while (original.Count > newContents.Count())
```

Comment:
I would hope the C# compiler is smart enough to cache this value as we don't modify `newContents` here, but now that it's an enumerable, it could be more expensive to call the `Count()` method. I think, iirc, since the underlying type is still just a collection it should be a cheap call. This is where my boundary of the underbelly of how C# functions gets more fuzzy.

That said, may be better to explicitly save this off in a local scoped var and reference here and in the if below.

Reply:
mmk done in 34282b3